server/channels/app/server.go

```diff
@@ -1152,7 +1152,19 @@ func (a *App) OriginChecker() func(*http.Request) bool {
 
 		return utils.OriginChecker(allowed)
 	}
-	return nil
+
+	// Overriding the default origin checker
+	return func(r *http.Request) bool {
+		origin := r.Header["Origin"]
+		if len(origin) == 0 {
+			return true
+		}
+		u, err := url.Parse(origin[0])
+		if err != nil {
+			return false
+		}
+		return strings.EqualFold(u.Host, r.Host) && strings.EqualFold(u.Scheme, r.URL.Scheme)
```

Comment:
If SSL termination is done in a reverse proxy (nginx), will `r.URL.Scheme` still be HTTPS here?

Reply:
Ah yeah good point. What do you suggest?

Reply:
Use the site URL I guess? This is the point why this issue was so tricky to begin with though. I don't know if we'll run into issues with misconfigured site URLs.

Reply:
Interesting though, it looks like we are completely ignoring `WebsocketURL` in the code. It's not being used anywhere AFAI can see.

Reply:
The websocket URL is used by the client, but the site URL is what we care about here.

Reply:
PTAL